[Merged by Bors] - Add docs on how to connect to Trino (#409)
stackabletech/trino-operator: docs/modules/trino/pages/usage_guide/connect_to_trino.adoc

= Connect to Trino

Trino offers a SQL endpoint via a HTTP Api.
A https://trino.io/docs/current/client/jdbc.html[Open source JDBC driver] does exist as well as a proprietary https://docs.starburst.io/data-consumer/clients/odbc.html[ODBC driver].

Multiple SQL clients can be used to connect to Trino as described below.

== Perquisites
The Trino server must be reachable from your local computer.

> It is normally secured via HTTPS with a certificated provided by a xref:secret-operator:secretclass.adoc[SecretClass].

Reviewer: ```suggestion
It is typically secured via HTTPS with a certificated provided by a xref:secret-operator:secretclass.adoc[SecretClass].
```

As most setups use a self-signed certificate, the easiest way is to disable the Trino cert validation as the following guide will do.

> A production-ready solution would be to extract the `ca.crt` from the SecretClass and provide it to the Trino client.

Reviewer: > A production-ready solution would be to ...

We should explain that part eventually too.

== Connect with trino-cli
Please have a look at the https://trino.io/docs/current/client/cli.html[official trino-cli docs].
Please don't forget to specify the `--insecure` flag, so you command looks something like:

[source,console]
----
java -jar ~/Downloads/trino-cli-403-executable.jar --server https://85.215.195.29:8443 --user admin --password --insecure
----

== Connect with DBeaver

> https://dbeaver.io[DBeaver] is a free multi-platform database tool for all people who need to work with databases.

Reviewer: ```suggestion
https://dbeaver.io[DBeaver] is a free multi-platform database tool for anyone who needs to work with databases.
```

Author: Copied that quote from their website but I'm fine with the suggestion ^^

It is installed locally and can connect to Trino and offers a convenient UI to explore and work with Trino.

First of you need to click on the `New Database Connection` icon, select `Other` -> `Trino`

image::connect-with-dbeaver-1.png[]

> Afterwards you need to enter the `Host` as well as the `Port` as well as `Username` and `Password`.

Reviewer: ```suggestion
Afterwards you need to enter the `Host` and `Port` as well as `Username` and `Password`.
```

image::connect-with-dbeaver-2.png[]

After entering the details you need to switch to the tab called `Driver properties`.
First action is to click on the `SSL` driver property and enter the value `true`.

> Additionally you need to click on the green plus icon on the bottom left corner to add a User Property called `SSLVerification`.

Reviewer: ```suggestion
Additionally you need to click on the green plus icon in the bottom left corner to add a User Property called `SSLVerification`.
```
